src/jace/translated_jaxpr_sdfg.py

```diff
@@ -0,0 +1,229 @@
+        for lowering the SDFG.
+        In DaCe the return values are allocated on a per `CompiledSDFG` basis. Thus
+        every call to a compiled SDFG will override the value of the last call, in JaCe
+        the memory is allocated on every call. In addition scalars are returned as
+        arrays of length one.
+    """
+
+    csdfg: compiled_sdfg.CompiledSDFG
+    sdfg: dace.SDFG
+    inp_names: tuple[str, ...]
+    out_names: tuple[str, ...]
+
+    def __init__(
+        self,
+        csdfg: compiled_sdfg.CompiledSDFG,
+        inp_names: tuple[str, ...],
+        out_names: tuple[str, ...],
+    ) -> None:
+        self.csdfg = csdfg
+        self.sdfg = self.csdfg.sdfg
+        self.inp_names = inp_names
+        self.out_names = out_names
+
+    def __call__(
+        self,
+        flat_call_args: Sequence[Any],
+    ) -> list[np.ndarray] | None:
+        """
+        Run the compiled SDFG using the flattened input.
+
+        The function will not perform flattening of its input nor unflattening of
+        the output.
+
+        Args:
+            csdfg: The compiled SDFG to call.
+            flat_call_args: Flattened input arguments.
+        """
+        if len(self.inp_names) != len(flat_call_args):
+            # Either error or static arguments are not removed.
+            raise RuntimeError("Wrong number of arguments.")
+
+        sdfg_call_args: dict[str, Any] = {}
+        for in_name, in_val in zip(self.inp_names, flat_call_args):
+            # TODO(phimuell): Implement a stride matching process.
+            if util.is_jax_array(in_val):
+                if not util.is_fully_addressable(in_val):
+                    raise ValueError(f"Passed a not fully addressable Jax array as '{in_name}'")
+                in_val = in_val.__array__()  # noqa: PLW2901  # Jax arrays do not expose the __array_interface__.
+            sdfg_call_args[in_name] = in_val
+
+        arrays = self.sdfg.arrays
+        for out_name, sdfg_array in ((out_name, arrays[out_name]) for out_name in self.out_names):
+            if out_name in sdfg_call_args:
+                if util.is_jax_array(sdfg_call_args[out_name]):
+                    raise ValueError("Passed an immutable Jax array as output.")
+            else:
+                sdfg_call_args[out_name] = dace_data.make_array_from_descriptor(sdfg_array)
+
+        assert len(sdfg_call_args) == len(self.csdfg.argnames), (
+            "Failed to construct the call arguments,"
+            f" expected {len(self.csdfg.argnames)} but got {len(flat_call_args)}."
+            f"\nExpected: {self.csdfg.argnames}\nGot: {list(sdfg_call_args.keys())}"
+        )
+
+        # Calling the SDFG
+        with dace.config.temporary_config():
+            dace.Config.set("compiler", "allow_view_arguments", value=True)
+            self.csdfg(**sdfg_call_args)
+
+        if self.out_names:
+            return [sdfg_call_args[out_name] for out_name in self.out_names]
+        return None
```

Comment:
1. It could be a list, the reason I put it inside a list is because JAX put flattened arguments inside a list, this is not a collection of return values but a list that is returned.
2. The reason for `None` is for handling inplace modification of return value, which is needed for ICON stencils.